### PR TITLE
made _convolve_data public and renamed filter_data

### PR DESCRIPTION
**Pull request by johnnygreco**
I've made _convolve_data public and renamed it filter_data. This will serve as a convenience function for users who would like to inspect their images at each step of their pipeline. This is a follow-up to PR  #397. 

Thanks

### CODE REVIEW

File: photutils/detection/deblend.py [hunk 1]
```diff
@@ -112,7 +112,7 @@ def deblend_sources(data, segment_img, npixels, filter_kernel=None,
         labels = segment_img.labels
     labels = np.atleast_1d(labels)
 
-    data = _convolve_data(data, filter_kernel, mode='constant',
+    data = filter_data(data, filter_kernel, mode='constant',
                           fill_value=0.0)
```
Comment (larrybradley) [inline photutils/detection/deblend.py]: The indentation on this line needs to be adjusted.  The "f" in `fill_value` should be directly below the "d" in `data` in the line above.

File: photutils/detection/findstars.py [hunk 2]
```diff
@@ -426,7 +426,7 @@ def _findobjs(data, threshold, kernel, min_separation=None,
                     x_kernradius:x_kernradius + data.shape[1]] = data
         data = data_padded
 
-    convolved_data = _convolve_data(data, kernel.kern, mode='constant',
+    convolved_data = filter_data(data, kernel.kern, mode='constant',
                                     fill_value=0.0, check_normalization=False)
```
Comment (larrybradley) [inline photutils/detection/findstars.py]: Fix indentation here.

File: photutils/segmentation.py [hunk 3]
```diff
@@ -1687,7 +1687,7 @@ def source_properties(data, segment_img, error=None, mask=None,
 
     # filter the data once, instead of repeating for each source
     if filter_kernel is not None:
-        filtered_data = _convolve_data(data, filter_kernel, mode='constant',
+        filtered_data = filter_data(data, filter_kernel, mode='constant',
                                        fill_value=0.0,
                                        check_normalization=True)
```
Comment (larrybradley) [inline photutils/segmentation.py]: Again for the above two lines.

File: photutils/utils/convolution.py [hunk 4]
```diff
@@ -6,8 +6,9 @@
 from astropy.convolution import Kernel2D
 from astropy.utils.exceptions import AstropyUserWarning
 
+__all__ = ['filter_data']
 
-def _convolve_data(data, kernel, mode='constant', fill_value=0.0,
+def filter_data(data, kernel, mode='constant', fill_value=0.0,
                    check_normalization=False):
```
Comment (larrybradley) [inline photutils/utils/convolution.py]: Again here.

**Comment by larrybradley** [comment]
@johnnygreco This looks good, thanks!  There are just a few minor indentations that need to be fixed (see inline comments).

**Comment by larrybradley** [comment]
Thanks, @johnnygreco.

**Comment by johnnygreco** [comment]
my pleasure!  thank you.

**Pull request merged**